### PR TITLE
Migrate Axes Display

### LINKED ISSUE #93: Migrate AxesDisplay

### PR DESCRIPTION
**Pull request by Martin-Idel**
Closes #93 

- Migrate display and fix linters
- Add visual test for axes display.
- Do not display axes when transformation fails
- Small refactoring of axes class for c++14

### CODE REVIEW

File: rviz_default_plugins/plugins_description.xml [hunk 1]
```diff
@@ -2,6 +2,17 @@
 
   <!-- Displays plugins -->
 
+  <class
+          name="rviz_default_plugins/Axes"
```
Comment (jacobperron) [inline rviz_default_plugins/plugins_description.xml]: nit: the attributes look over-indented

**Review by jacobperron — COMMENTED** [review]
Sorry for taking so long to get to this.

LGTM :+1: After a rebase I'll trigger CI.

**Comment by Martin-Idel** [comment]
Rebase done.

I also took the liberty to tweak the readme: The "Oculus" display isn't part of this repository and the Interactive Marker display as well as message filters have been ported. On the other hand, stereo functionality is currently not enabled, so that should be put on the list. 

I can revert those changes if you want to, of course.

**Comment by jacobperron** [comment]
I noticed that with the default value, `FIXED_FRAME`, it is unable to resolve the transformation even if the global fixed frame is available. I guess this is an issue to be resolved in the frame transformer. 
We can ticket it separately if you agree.

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8245/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4212/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6727/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8135/)

Edit: re-triggered with dbe74f6

File: rviz_default_plugins/src/rviz_default_plugins/displays/axes/axes_display.cpp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  axes_->set(length_property_->getFloat(), radius_property_->getFloat());
+  context_->queueRender();
+}
+
+void AxesDisplay::update(float dt, float ros_dt)
+{
+  (void) dt;
+  (void) ros_dt;
+
+  std::string frame = frame_property_->getStdString();
```
Comment (jacobperron) [inline rviz_default_plugins/src/rviz_default_plugins/displays/axes/axes_display.cpp]: Here's the bug, it should be `getFrameStd()` instead:
```suggestion
  std::string frame = frame_property_->getFrameStd();
```

Comment (jacobperron) [reply rviz_default_plugins/src/rviz_default_plugins/displays/axes/axes_display.cpp]: See dbe74f6

**Comment by Martin-Idel** [comment]
Thanks for catching this. I only retested with the visual test and some frame transformations. Looks good to me.

**Pull request merged**